pose_prediction/plugin.cpp

```diff
@@ -56,34 +57,34 @@ class pose_prediction_impl : public pose_prediction {
 
 
     // future_time: An absolute timepoint in the future
-    virtual fast_pose_type get_fast_pose(time_type future_timestamp) const override {
+    virtual fast_pose_type get_fast_pose(time_point future_timestamp) const override {
         switchboard::ptr<const pose_type> slow_pose = _m_slow_pose.get_ro_nullable();
         if (slow_pose == nullptr) {
             // No slow pose, return 0
             return fast_pose_type{
                 correct_pose(pose_type{}),
-                std::chrono::system_clock::now(),
+                _m_clock->now(),
                 future_timestamp,
             };
         }
 
         switchboard::ptr<const imu_raw_type> imu_raw = _m_imu_raw.get_ro_nullable();
         if (imu_raw == nullptr) {
 #ifndef NDEBUG
-            printf("FAST POSE IS SLOW POSE!");
+            printf("FAST POSE IS SLOW POSE!\n");
 #endif
             // No imu_raw, return slow_pose
             return fast_pose_type{
-                correct_pose(*slow_pose),
-                std::chrono::system_clock::now(),
-                future_timestamp,
+                .pose = correct_pose(*slow_pose),
+                .predict_computed_time = _m_clock->now(),
+                .predict_target_time = future_timestamp,
             };
         }
 
         // slow_pose and imu_raw, do pose prediction
 
-        double dt = std::chrono::duration_cast<std::chrono::nanoseconds>(future_timestamp - std::chrono::system_clock::now()).count();
-        std::pair<Eigen::Matrix<double,13,1>, time_type> predictor_result = predict_mean_rk4(dt/NANO_SEC);
+        double dt = duration2double(future_timestamp - imu_raw->imu_time);
+        std::pair<Eigen::Matrix<double,13,1>, time_point> predictor_result = predict_mean_rk4(dt);
```

Comment:
Just want to confirm that duration2double will return fractions of seconds because the type of `Rep` in chrono::duration's template class is a double?

Reply:
Yup